client/src/Components/Product/ProductMainView.test.js

```diff
@@ -0,0 +1,51 @@
+import React from "react"
+import { render, screen } from "@testing-library/react"
+import "@testing-library/jest-dom"
+import ProductMainView from "./ProductMainView.jsx"
+import { ProductsContext } from "../../contexts/ProductsContext.jsx"
+
+describe("Product", () => {
+  test("renders Product component", () => {
+    render(<ProductMainView />)
+  })
+})
+
+describe("Product Title", () => {
+  test("Finds the Project Title", () => {
+    render(<ProductMainView />)
+    expect(screen.getByRole("header")).toBeInTheDocument()
+  })
+})
+
+describe("Product Price", () => {
+  test("Finds the price", () => {
+    render(<ProductMainView />)
+    expect(screen.getByText("price")).toBeInTheDocument()
+  })
+})
+
+describe("Ratings", () => {
+  test("Ratings", () => {
+    render(<ProductMainView />)
+    expect(screen.getByText("ratings")).toBeInTheDocument()
+  })
+})
+
+describe("Bag exists", () => {
+  test("Finds the bag", () => {
+    render(<ProductMainView />)
+    expect(screen.getByRole("Bag")).toBeInTheDocument() ||
+      expect(screen.getByRole("Bag")).toBeInTheDocument()
+  })
+})
+
+describe("Quantity selector works", () => {
+  test("Quantity selector works", () => {
+    render(<ProductMainView />)
+    expect(screen.getByText("button")).toBeInTheDocument()
+  })
+})
```

Comment:
Don't spend too much effort on testing if it wastes time, but here are a few things to note:
 - Refer to the [Roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques) mdn docs
 - You can use a `beforeEach` function to render the component
 - You can wrap all of the tests in the `describe` function

## Suggestion:
```suggestion
describe("Product", () => {
   beforeEach(() => {
     render(<ProductMainView />)
   })

  test("Finds the Project Title", () => {
    expect(screen.getByRole("header")).toBeInTheDocument()
  })

  test("Finds the price", () => {
    expect(screen.getByText("price")).toBeInTheDocument()
  })

  test("Ratings", () => {
    expect(screen.getByText("ratings")).toBeInTheDocument()
  })

  test("Finds the bag", () => {
      expect(screen.getByRole("Bag")).toBeInTheDocument() ||
      expect(screen.getByRole("Bag")).toBeInTheDocument()
  })

  test("Quantity selector works", () => {
    expect(screen.getByText("button")).toBeInTheDocument()
  })
})
```

## Container:
An alternative for screen.getByX is using containner. This is a hacky work-around, and I think it could lead to testing anti-patterns. I think it's bad to test for 'div.someClass' becuase your psuedo-writing the code through the test, but doing twice the work.
You can do DOM selector queries like this:
```js
describe("Product", () => {
  let container
   beforeEach(() => {
     const container = render(<ProductMainView />).contianer // yikes
   })

   test("should render the answer to life", () => {
    expect(container.querySelector('h3.yourClass').textContent).toBe('42')
  })
})
```